scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -45,6 +48,10 @@ def _get_proxy(self, url, orig_type):
     def process_request(self, request, spider):
         # ignore if proxy is already set
         if 'proxy' in request.meta:
+            # extract auth info if present
```

Comment:
I would also test to not overwrite an already set `"Proxy-Authentication"` header